bitnami/redis-cluster/values-production.yaml

```diff
@@ -45,6 +45,10 @@ cluster:
   ## Enable the creation of a Job that will execute the proper command to create the Redis Cluster.
   ##
   init: true
+  ## Helm hooks for the "helm.sh/hook" annotation.
+  ## Leave empty to run the cluster creation Job without using Helm hooks.
```

Comment:
I think the cluster will not be created if the hook is deleted.  Also, the job will be part of the release if you remove the hooks, so you will be unable to upgrade the chart again, as a Job is an inmutable object

Reply:
You are right, I pushed another change that should solve this issue

Reply:
I tested my changes without using helm hook. Here is the output for the init-cluster job for `helm install`:
```
2020-06-22T13:32:01.777054139Z [38;5;6m [38;5;5m13:32:01.77 [0m
2020-06-22T13:32:01.77838495Z [38;5;6m [38;5;5m13:32:01.77 [0m[1mWelcome to the Bitnami redis-cluster container[0m
2020-06-22T13:32:01.77949896Z [38;5;6m [38;5;5m13:32:01.77 [0mSubscribe to project updates by watching [1mhttps://github.com/bitnami/bitnami-docker-redis-cluster[0m
2020-06-22T13:32:01.780924572Z [38;5;6m [38;5;5m13:32:01.78 [0mSubmit issues and feature requests at [1mhttps://github.com/bitnami/bitnami-docker-redis-cluster/issues[0m
2020-06-22T13:32:01.782118482Z [38;5;6m [38;5;5m13:32:01.78 [0m
2020-06-22T13:32:01.783465793Z [38;5;6m [38;5;5m13:32:01.78 [0m[38;5;2mINFO [0m ==> ** Starting Redis setup **
2020-06-22T13:32:01.804072465Z [38;5;6m [38;5;5m13:32:01.80 [0m[38;5;2mINFO [0m ==> Initializing Redis...
2020-06-22T13:32:01.874783557Z [38;5;6m [38;5;5m13:32:01.87 [0m[38;5;2mINFO [0m ==> ** Redis setup finished! **
2020-06-22T13:32:01.874845558Z 
2020-06-22T13:32:02.030575261Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:32:02.030953864Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:32:03.118405862Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:32:03.118741665Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:32:04.14404034Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:32:04.144346642Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:32:05.167117593Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:32:05.167416095Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:32:06.188682131Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:32:06.188981233Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:32:07.207476343Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:32:07.207838146Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:32:08.228212968Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:32:08.228574671Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...

[ TRUNCATED ]

2020-06-22T13:34:31.819882512Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:34:31.820183414Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:32.845143024Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:34:32.845147724Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:33.860433654Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Name or service not known
2020-06-22T13:34:33.860780156Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:34.880867023Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: No address associated with hostname
2020-06-22T13:34:34.881189425Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:35.901579992Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: No address associated with hostname
2020-06-22T13:34:35.901912994Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:36.910603065Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Connection refused
2020-06-22T13:34:36.910923468Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:37.919715837Z Could not connect to Redis at chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless:6379: Connection refused
2020-06-22T13:34:37.919836738Z Node chart-1592832717-redis-cluster-0.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:38.942814319Z Could not connect to Redis at chart-1592832717-redis-cluster-1.chart-1592832717-redis-cluster-headless:6379: Connection refused
2020-06-22T13:34:38.943314623Z Node chart-1592832717-redis-cluster-1.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:39.95097738Z Could not connect to Redis at chart-1592832717-redis-cluster-1.chart-1592832717-redis-cluster-headless:6379: Connection refused
2020-06-22T13:34:39.951279082Z Node chart-1592832717-redis-cluster-1.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:34:42.060871242Z >>> Performing hash slots allocation on 6 nodes...
2020-06-22T13:34:42.060896542Z Master[0] -> Slots 0 - 5460
2020-06-22T13:34:42.060902742Z Master[1] -> Slots 5461 - 10922
2020-06-22T13:34:42.060908042Z Master[2] -> Slots 10923 - 16383
2020-06-22T13:34:42.060922042Z Adding replica 10.244.3.7:6379 to 10.244.1.11:6379
2020-06-22T13:34:42.060925842Z Adding replica 10.244.0.11:6379 to 10.244.2.8:6379
2020-06-22T13:34:42.060928942Z Adding replica 10.244.1.12:6379 to 10.244.0.10:6379
2020-06-22T13:34:42.060931943Z M: 8c553be3712532c581e462a58898b73b79292fcf 10.244.1.11:6379
2020-06-22T13:34:42.060934843Z    slots:[0-5460] (5461 slots) master
2020-06-22T13:34:42.060937843Z M: 115a65d5b4fb14aa4a15e7f7048bcd72c8b28c55 10.244.2.8:6379
2020-06-22T13:34:42.060940843Z    slots:[5461-10922] (5462 slots) master
2020-06-22T13:34:42.060943543Z M: ee6f2885760d65291a7994410ff34e35e00e7c4b 10.244.0.10:6379
2020-06-22T13:34:42.060946343Z    slots:[10923-16383] (5461 slots) master
2020-06-22T13:34:42.060949443Z S: c4aed3fa1f171c5e8937b3b1acae78318336ca80 10.244.1.12:6379
2020-06-22T13:34:42.060952243Z    replicates ee6f2885760d65291a7994410ff34e35e00e7c4b
2020-06-22T13:34:42.060962743Z S: 5ee583ff90febda12aa33ff156dc3542189ede30 10.244.3.7:6379
2020-06-22T13:34:42.060965843Z    replicates 8c553be3712532c581e462a58898b73b79292fcf
2020-06-22T13:34:42.060968743Z S: d4a40ecbb00f3421acf501dfbbbd4660bab446f4 10.244.0.11:6379
2020-06-22T13:34:42.060971543Z    replicates 115a65d5b4fb14aa4a15e7f7048bcd72c8b28c55
2020-06-22T13:34:42.060974343Z >>> Nodes configuration updated
2020-06-22T13:34:42.060978743Z >>> Assign a different config epoch to each node
2020-06-22T13:34:42.060983743Z >>> Sending CLUSTER MEET messages to join the cluster
2020-06-22T13:34:42.060988843Z Waiting for the cluster to join
2020-06-22T13:34:45.096334186Z ...
2020-06-22T13:34:45.096367986Z >>> Performing Cluster Check (using node 10.244.1.11:6379)
2020-06-22T13:34:45.096372586Z M: 8c553be3712532c581e462a58898b73b79292fcf 10.244.1.11:6379
2020-06-22T13:34:45.096376786Z    slots:[0-5460] (5461 slots) master
2020-06-22T13:34:45.096381586Z    1 additional replica(s)
2020-06-22T13:34:45.096386386Z S: 5ee583ff90febda12aa33ff156dc3542189ede30 10.244.3.7:6379
2020-06-22T13:34:45.096391486Z    slots: (0 slots) slave
2020-06-22T13:34:45.096396386Z    replicates 8c553be3712532c581e462a58898b73b79292fcf
2020-06-22T13:34:45.096401586Z S: c4aed3fa1f171c5e8937b3b1acae78318336ca80 10.244.1.12:6379
2020-06-22T13:34:45.096406686Z    slots: (0 slots) slave
2020-06-22T13:34:45.096411586Z    replicates ee6f2885760d65291a7994410ff34e35e00e7c4b
2020-06-22T13:34:45.096422686Z S: d4a40ecbb00f3421acf501dfbbbd4660bab446f4 10.244.0.11:6379
2020-06-22T13:34:45.096425986Z    slots: (0 slots) slave
2020-06-22T13:34:45.096428886Z    replicates 115a65d5b4fb14aa4a15e7f7048bcd72c8b28c55
2020-06-22T13:34:45.096431786Z M: ee6f2885760d65291a7994410ff34e35e00e7c4b 10.244.0.10:6379
2020-06-22T13:34:45.096434887Z    slots:[10923-16383] (5461 slots) master
2020-06-22T13:34:45.096437787Z    1 additional replica(s)
2020-06-22T13:34:45.096440587Z M: 115a65d5b4fb14aa4a15e7f7048bcd72c8b28c55 10.244.2.8:6379
2020-06-22T13:34:45.096443987Z    slots:[5461-10922] (5462 slots) master
2020-06-22T13:34:45.096446887Z    1 additional replica(s)
2020-06-22T13:34:45.096449787Z [OK] All nodes agree about slots configuration.
2020-06-22T13:34:45.096452487Z >>> Check for open slots...
2020-06-22T13:34:45.096455387Z >>> Check slots coverage...
2020-06-22T13:34:45.096458187Z [OK] All 16384 slots covered.
2020-06-22T13:34:45.122654996Z Cluster correctly created
```

and here is the output for `helm upgrade`:
```
2020-06-22T13:41:18.895879352Z [38;5;6m [38;5;5m13:41:18.89 [0m
2020-06-22T13:41:18.897110762Z [38;5;6m [38;5;5m13:41:18.89 [0m[1mWelcome to the Bitnami redis-cluster container[0m
2020-06-22T13:41:18.898387871Z [38;5;6m [38;5;5m13:41:18.89 [0mSubscribe to project updates by watching [1mhttps://github.com/bitnami/bitnami-docker-redis-cluster[0m
2020-06-22T13:41:18.89964308Z [38;5;6m [38;5;5m13:41:18.89 [0mSubmit issues and feature requests at [1mhttps://github.com/bitnami/bitnami-docker-redis-cluster/issues[0m
2020-06-22T13:41:18.900996291Z [38;5;6m [38;5;5m13:41:18.90 [0m
2020-06-22T13:41:18.902369101Z [38;5;6m [38;5;5m13:41:18.90 [0m[38;5;2mINFO [0m ==> ** Starting Redis setup **
2020-06-22T13:41:18.923186455Z [38;5;6m [38;5;5m13:41:18.92 [0m[38;5;2mINFO [0m ==> Initializing Redis...
2020-06-22T13:41:18.98699123Z [38;5;6m [38;5;5m13:41:18.98 [0m[38;5;2mINFO [0m ==> ** Redis setup finished! **
2020-06-22T13:41:18.98703373Z 
2020-06-22T13:41:29.321311513Z Could not connect to Redis at chart-1592832717-redis-cluster-5.chart-1592832717-redis-cluster-headless:6379: No route to host
2020-06-22T13:41:29.321682016Z Node chart-1592832717-redis-cluster-5.chart-1592832717-redis-cluster-headless not ready, waiting for all the nodes to be ready...
2020-06-22T13:41:30.349536847Z [ERR] Node 10.244.1.15:6379 is not empty. Either the node already knows other nodes (check with CLUSTER NODES) or contains some key in database 0.
2020-06-22T13:41:30.376028144Z Cluster correctly created
```

It looks like the init-cluster is waiting for all the nodes to come up as expected.

Reply:
I just realized that if the `--wait` flag is not specified, the `post-install` and `post-upgrade` hooks won't wait for all the resources to be ready. I must have misinterpreted the [doc](https://helm.sh/docs/topics/charts_hooks/#hooks-and-the-release-lifecycle) when I first read it. I will revert my changes regarding the `helmHooks` parameter and repurpose my PR for only the unrelated fixes I made.

Reply:
In spite of the message `Cluster correctly created` I think it was not properly created because the node was not empty, the error just before that line indicates that the cluster failed to create. We should improve that behaviour of the script that is due to the `|| true`.

Reply:
I should open an issue in the docker image repository for that: https://github.com/bitnami/bitnami-docker-redis-cluster/blob/master/6.0/debian-10/rootfs/opt/bitnami/scripts/librediscluster.sh, right?